lib/PlatoonService/src/V2VCommManager.cpp

```diff
@@ -0,0 +1,534 @@
+    DeserializationError                        error = deserializeJson(jsonPayload, payload.c_str());
+
+    if (DeserializationError::Ok != error)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        /* Send vehicle heartbeat. */
+        JsonVariant                                 jsonTimestamp = jsonPayload["timestamp"]; /* Timestamp [ms]. */
+        StaticJsonDocument<JSON_HEARTBEAT_MAX_SIZE> heartbeatDoc;
+        String                                      heartbeatPayload;
+
+        if (false == jsonTimestamp.isNull())
+        {
+            heartbeatDoc["id"]        = m_vehicleId;
+            heartbeatDoc["timestamp"] = jsonTimestamp.as<uint32_t>();
+            heartbeatDoc["status"]    = m_vehicleStatus;
+
+            if (0U == serializeJson(heartbeatDoc, heartbeatPayload))
+            {
+                LOG_ERROR("Failed to serialize heartbeat.");
+            }
+            else if (false == m_mqttClient.publish(m_heartbeatResponseTopic, false, heartbeatPayload))
+            {
+                LOG_ERROR("Failed to publish MQTT message to %s.", m_heartbeatResponseTopic.c_str());
+            }
+            else
+            {
+                LOG_DEBUG("Sent heartbeat: %s", heartbeatPayload.c_str());
+            }
+        }
+    }
+}
+
+void V2VCommManager::vehicleHeartbeatTopicCallback(const String& payload)
+{
+    /* Deserialize payload. */
+    StaticJsonDocument<JSON_HEARTBEAT_MAX_SIZE> jsonPayload;
+    DeserializationError                        error = deserializeJson(jsonPayload, payload.c_str());
+
+    if (DeserializationError::Ok != error)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        JsonVariant jsonId        = jsonPayload["id"];        /* Vehicle ID. */
+        JsonVariant jsonTimestamp = jsonPayload["timestamp"]; /* Timestamp [ms]. */
+        JsonVariant jsonStatus    = jsonPayload["status"];    /* Vehicle status. */
+
+        if ((false == jsonId.isNull()) && (false == jsonTimestamp.isNull()) && (false == jsonStatus.isNull()))
+        {
+            uint8_t  id        = jsonId.as<uint8_t>();
+            uint32_t timestamp = jsonTimestamp.as<uint32_t>();
+            uint8_t  status    = jsonStatus.as<uint8_t>();
+
+            if (PLATOON_LEADER_ID == id)
+            {
+                /* This is me, the leader! */
+            }
+            else if (id > NUMBER_OF_FOLLOWERS)
+            {
+                LOG_ERROR("Received heartbeat from unknown vehicle %d.", id);
+            }
+            else if (timestamp != m_lastPlatoonHeartbeatTimestamp)
+            {
+                LOG_ERROR("Received heartbeat from vehicle %d with timestamp %d, expected %d.", id, timestamp,
+                          m_lastPlatoonHeartbeatTimestamp);
+            }
+            else
+            {
+                uint8_t   followerArrayIndex = id - 1U; /*Already checked id != 0U */

```

Comment:
What about larger than the array? ;-)

Reply:
Also checked with `if (id > NUMBER_OF_FOLLOWERS)`